### PR TITLE
CVE-2010-0425

### PR DESCRIPTION
**Pull request by bjt6087**

### CODE REVIEW

File: cves/CVE-2010-0425.yml [hunk 1]
```diff
@@ -54,7 +59,7 @@ fixes:
      note:
 vccs:
   - commit: 30ab960e20117ee6002a3297791427023440e0a7
-    note:
+    note: initial implementation of the file.
   - commit:
     note:
 upvotes_instructions: |
```
Comment (mesternefeld) [inline cves/CVE-2010-0425.yml:65]: This one is very interesting to me. I'll give it 2 upvotes!

File: cves/CVE-2010-0425.yml [hunk 2]
```diff
@@ -97,8 +102,9 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
-  answer:
-  date:
+  answer: The vulnerability was found by Brett Gervasoni of Sense of Security
+    who also proposed a fix for the vulnerability.
```
Comment (wrs1492) [inline cves/CVE-2010-0425.yml]: Was there any exploits created for this specific vulnerability, or was it fixed before attackers could abuse it?

File: cves/CVE-2010-0425.yml [hunk 3]
```diff
@@ -184,8 +201,10 @@ lessons:
     applies:
     note:
   yagni:
-    applies:
-    note:
+    applies: true
+    note: I feel like this lesson kind of applies here since the piece of code
```
Comment (wrs1492) [inline cves/CVE-2010-0425.yml]: "feel like" and "kind of" should not both be here, I personally would put "Yagni applies here" as I feel like this is the perfect example of Yagni.

File: cves/CVE-2010-0425.yml [hunk 4]
```diff
@@ -202,4 +221,14 @@ mistakes:
...
-  answer:
+  answer: I feel like this vulnerability was due to the developers overlooking
+    it and not quite understanding what was happening in a piece of their code.
+    The fix for the vulnerability was simply deleting 2 lines of code which
+    unloaded the ISAPI.dll module in the middle of a request which created
+    oprhaned callback pointers. Based on this fix it seems that the
+    unloading of the ISAPI.dll method call at that point in the code was
+    completely unecessary. I feel like this vulnerability could have been
+    caught and addressed earlier if the code was tested/reviewed more
+    thoroughly. The biggest takeaway for me from this vulnerability is even
```
Comment (wrs1492) [inline cves/CVE-2010-0425.yml]: I think there are probably better takeaways from this vulnerability, as that lesson is pretty generic and kind of obvious. I would instead consider why this vulnerability took so long to be found/fixed.

**Comment by wrs1492** [comment]
+2 upvotes

File: cves/CVE-2010-0425.yml (hunk 4 above)
Comment (bjt6087) [reply cves/CVE-2010-0425.yml]: Changed the lessons learned in latest commit.

File: cves/CVE-2010-0425.yml (hunk 3 above)
Comment (bjt6087) [reply cves/CVE-2010-0425.yml]: Fixed wording in latest commit.

File: cves/CVE-2010-0425.yml (hunk 2 above)
Comment (bjt6087) [reply cves/CVE-2010-0425.yml]: From what I researched I saw no evidence of an exploit being created for this vulnerability so I believe that would mean it was fixed before attackers could abuse it. I added this information to the CVE in the latest commit.

**Pull request merged**